drivers/adc/adc_npcx.c

```diff
@@ -37,6 +38,12 @@ LOG_MODULE_REGISTER(adc_npcx, CONFIG_ADC_LOG_LEVEL);
 #define NPCX_ADC_CHN_CONVERSION_MODE	0
 #define NPCX_ADC_SCAN_CONVERSION_MODE	1
 
+/* ADC threshold detector number */
+#define NPCX_ADC_THRESHOLD_COUNT   DT_INST_PROP(0, threshold_count)
```

Comment:
Non-blocking comment: Can you add `threshold_count` to `adc_npcx_config`? Hiding this behind a macro implies that this is a constant for all NPCX types.  This can be done as a separate PR.

Also, I think the `threshold_reg_offset` property should also get moved into `adc_npcx_config`.  The `THRCTL` macro should also get moved into this file.

Reply:
I suggest a single PR broken up into 2 commits.